core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -216,7 +216,8 @@ class LogManager(logDirs: Array[File],
       maxProducerIdExpirationMs = maxPidExpirationMs,
       scheduler = scheduler,
       time = time,
-      brokerTopicStats = brokerTopicStats)
+      brokerTopicStats = brokerTopicStats,
+      logDirFailureChannel = logDirFailureChannel)
```

Comment:
This should have been a compiler error in the first place. The default that we have `null` as a default argument in `Log.apply` doesn't seem right. What's the motivation for that?